packages/core/lib/modules/accounts.js

```diff
@@ -2,63 +2,126 @@ import {WalletRpc} from '../client/wallet-rpc';
 import {UtilCryptoRpc} from '../client/util-crypto-rpc';
 import {KeyringRpc} from '../client/keyring-rpc';
 import {ApiRpc} from '../client/api-rpc';
-import {Wallet} from './wallet';
-import { Errors } from '../errors';
-
-export type KeypairType = 'sr25519' | 'ed25519' |  'ecdsa';
+import {Wallet, WalletEvents} from './wallet';
+import {Errors} from '../errors';
+import {KeypairType, WalletDocument, DocumentType} from '../types';
+import {EventManager} from './event-manager';
 
 export type Account = {
-  id: string;
-  name: string;
-  type: KeypairType;
-  address: string;
-}
+  id: string,
+  name: string,
+  type: KeypairType,
+  address: string,
+  correlations: any[],
+};
+
+export const AccountsEvents = {
+  loaded: 'loaded',
+  accountCreated: 'account-added',
+  accountUpdated: 'account-updated',
+  accountRemoved: 'account-removed',
+};
 
 // TODO: Add events to accounts
 export class Accounts {
   accounts: Account[];
   wallet: Wallet;
+  eventManager: EventManager;
+  static DocumentFilters = {
+    mnemonicType: (item: WalletDocument) => item.type === 'Mnemonic',
+    currencyType: (item: WalletDocument) => item.type === 'Currency',
+  };
 
   constructor() {
     this.accounts = [];
     this.wallet = Wallet.getInstance();
+    this.eventManager = new EventManager();
+    this.eventManager.registerEvents(AccountsEvents);
   }
 
   async load() {
     this.accounts = await Wallet.getInstance().query({
-      type: 'account',
+      type: 'Address',
     });
-
+    this.eventManager.emit(AccountsEvents.loaded);
     return this.accounts;
   }
 
-  getBalance(address) {
-    return ApiRpc.getAccountBalance(accountId);
+  async exportAccount(accountId, password) {
+    return WalletRpc.exportAccount(accountId, password);
```

Comment:
should export (and import below) fire off an event?

Reply:
I don't see any use case for it

Reply:
you're assuming since these actions are being initiated directly by the UI they won't need any kind of event notification?